File: src/code-samples/langsmith/smithdb-migration/threads-stats-basic-after.ts

async function findThreadId(projectId: string): Promise<string> {
  const { Client } = await import("langsmith");
  const client = new Client();
  for await (const thread of client.threads.query({
    project_id: projectId,
    min_start_time: "2026-07-01T00:00:00Z",
    max_start_time: "2026-07-31T23:59:59Z",
    page_size: 1,
  })) {
    return thread.thread_id!;
  }
  throw new Error("no threads found");
}

// :snippet-start: threads-stats-basic-after-js
// :codegroup-tab: After
import { Client } from "langsmith";

const client = new Client();
const project = await client.readProject({ projectName: "default" });
let threadId = "<thread-id>";
// :remove-start:
threadId = await findThreadId(project.id);
// :remove-end:
const stats = await client.threads.stats(threadId, {
  session_id: project.id,
  selects: ["TURNS", "TOTAL_TOKENS", "TOTAL_COST"],
});
console.log(stats.turns, stats.total_tokens, stats.total_cost);
// :snippet-end:

File: src/code-samples/langsmith/smithdb-migration/threads-stats-basic-after.kt

///usr/bin/env jbang "$0" "$@" ; exit $?
//JAVA 21
//KOTLIN 2.2.0
//DEPS com.langchain.smith:langsmith-java:0.1.0-beta.15

// :snippet-start: threads-stats-basic-after-kt
// :codegroup-tab: After
// :remove-start:
import java.time.OffsetDateTime
// :remove-end:

import com.langchain.smith.client.LangsmithClient
import com.langchain.smith.client.okhttp.LangsmithOkHttpClient
import com.langchain.smith.models.sessions.SessionListParams
// :remove-start:
import com.langchain.smith.models.threads.ThreadQueryParams
// :remove-end:
import com.langchain.smith.models.threads.ThreadStatsParams
import kotlin.jvm.optionals.getOrNull

// :remove-start:
fun main() {
    if (System.getenv("LANGSMITH_API_KEY").isNullOrBlank()) {
        println("[smithdb-threads-stats-basic-after] Skipping (LANGSMITH_API_KEY is not set).")
        return
    }
// :remove-end:
val client: LangsmithClient = LangsmithOkHttpClient.fromEnv()

val project = client.sessions().list(
    SessionListParams.builder().name("default").limit(1L).build()
).items().first()

var threadId = "<thread-id>"
// :remove-start:
threadId = client.threads().query(
    ThreadQueryParams.builder()
        .projectId(project.id())
        .minStartTime(OffsetDateTime.parse("2026-07-01T00:00:00Z"))
        .maxStartTime(OffsetDateTime.parse("2026-07-31T23:59:59Z"))
        .build()
).items().first().threadId().get()
// :remove-end:

val stats = client.threads().stats(
    threadId,
    ThreadStatsParams.builder()
        .sessionId(project.id())
        .addSelect(ThreadStatsParams.Select.TURNS)
        .addSelect(ThreadStatsParams.Select.TOTAL_TOKENS)
        .addSelect(ThreadStatsParams.Select.TOTAL_COST)
        .build()
)
println("${stats.turns().getOrNull()} ${stats.totalTokens().getOrNull()} ${stats.totalCost().getOrNull()}")
// :remove-start:
}
// :remove-end:
// :snippet-end:
